### PR TITLE
fix date/time widget for field with time zones

### PR DESCRIPTION
**Pull request by 3nids**
this adds a new format of date/time to use Qt::ISODate format
the issue is that QDate::fromString does not recognize the t letter for the timezone in format
this a hack since any field_format needs to be tested if it corresponds to QgsDateTimeFieldFormatter::DEFAULT_ISO_FORMAT and in such case replace by Qt::ISODate instead of a format string

I am not really convinced by this patch.
Would like someone to review @m-kuhn @nyalldawson ? 

https://issues.qgis.org/issues/16657

### CODE REVIEW

**Comment by 3nids** [comment]
ready for review.
@haubourg would be great if you could jump in

File: src/core/fieldformatter/qgsdatetimefieldformatter.cpp [hunk 1]
```diff
@@ -22,6 +22,7 @@
 const QString QgsDateTimeFieldFormatter::DEFAULT_DATE_FORMAT = QStringLiteral( "yyyy-MM-dd" );
 const QString QgsDateTimeFieldFormatter::DEFAULT_TIME_FORMAT = QStringLiteral( "HH:mm:ss" );
 const QString QgsDateTimeFieldFormatter::DEFAULT_DATETIME_FORMAT = QStringLiteral( "yyyy-MM-dd HH:mm:ss" );
+const QString QgsDateTimeFieldFormatter::DEFAULT_ISO_FORMAT = QStringLiteral( "yyyy-MM-dd HH:mm:ss+t" );
```
Comment (m-kuhn) [inline src/core/fieldformatter/qgsdatetimefieldformatter.cpp]: Is there no need for a separated `ISO_DATE` and `ISO_DATE_TIME`?

**Comment by 3nids** [comment]
I added a bool in the config to know if the field format is ISO or not.
This should avoid string comparison in the code (it happens only once in the config) and has a better readability.
Also fix the issue of the display format of QDateTimeEdit being string only (no Qt::ISODate possible).

File: src/gui/editorwidgets/qgsdatetimeeditconfig.cpp [hunk 2]
```diff
@@ -23,6 +23,13 @@ QgsDateTimeEditConfig::QgsDateTimeEditConfig( QgsVectorLayer *vl, int fieldIdx,
 {
   setupUi( this );
 
+  mFieldFormatComboBox->clear();
+  mFieldFormatComboBox->addItem( tr( "Date" ), QgsDateTimeFieldFormatter::DATE_FORMAT );
+  mFieldFormatComboBox->addItem( tr( "Time" ), QgsDateTimeFieldFormatter::TIME_FORMAT );
+  mFieldFormatComboBox->addItem( tr( "Date time" ), QgsDateTimeFieldFormatter::DATETIME_FORMAT );
+  mFieldFormatComboBox->addItem( tr( "ISO date time" ), QgsDateTimeFieldFormatter::QT_ISO_FORMAT );
+  mFieldFormatComboBox->addItem( tr( "Custom" ), "" );
```
Comment (m-kuhn) [inline src/gui/editorwidgets/qgsdatetimeeditconfig.cpp]: `""` > `QString()`?

**Pull request merged**